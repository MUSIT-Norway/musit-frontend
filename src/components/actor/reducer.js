import { Observable } from 'rxjs/Rx';
import actions from "./actions";

const dataFromServer = [
  {
    id: 1,
    fn: 'Jarl parl'
  },
  {
    id: 2,
    fn: 'Line pine'
  },
  {
    id: 3,
    fn: 'Kjell pell'
  },
  {
    id: 4,
    fn: 'Stein sein'
  },
  {
    id: 5,
    fn: 'Knut lut'
  },
  {
    id: 6,
    fn: 'Ellen'
  }
]

const initialState = [];

const ActorReducer$ = Observable.of(() => initialState)
  .merge(
    actions.suggest$.map((update) => () => {
      if (!update) {
        return initialState;
      }
      return dataFromServer.filter(d => {
        const name = d.fn.toLowerCase();
        const query = update.value.toLowerCase();
        return name.indexOf(query) > -1
      })
    }),
    actions.clear$.map(() => () => initialState),
  );

export default Observable.merge(
  ActorReducer$.map(reducer => ["suggest", reducer]),
);