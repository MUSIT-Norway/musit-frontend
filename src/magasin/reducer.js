import { toFrontend } from './mapper';
import * as types from './types';

const initialState = {
  root: {
    data: {}
  }
};

const storageUnitGridReducer = (state = initialState, action = {}) => {
  switch (action.type) {
  case types.LOAD_SEVERAL:
    return {
      ...state,
      loading: true
    };
  case types.LOAD_SEVERAL_SUCCESS:
    return {
      ...state,
      loading: false,
      loaded: true,
      data: action.result
    };
  case types.LOAD_SEVERAL_FAIL:
    return {
      ...state,
      loading: false,
      loaded: false,
      error: action.error
    };
  case types.LOAD_ONE_SUCCESS:
    return {
      ...state,
      root: {
        ...state.root,
        loading: false,
        loaded: true,
        data: toFrontend(action.result)
      }
    };
  case types.LOAD_ONE_FAIL:
    return {
      ...state,
      root: {
        ...state.root,
        loading: false,
        loaded: false,
        error: action.error
      }
    };
  case types.CLEAR_ROOT: {
    return {
      ...state,
      root: {}
    };
  }
  case types.DELETE:
    return {
      ...state,
      loading: true
    };
  case types.DELETE_SUCCESS:
    return {
      ...state,
      loading: false,
      loaded: true,
      data: []
    };
  case types.DELETE_FAIL:
    return {
      ...state,
      loading: false,
      loaded: false,
      error: action.error
    };
  default:
    return state;
  }
};

export default storageUnitGridReducer;