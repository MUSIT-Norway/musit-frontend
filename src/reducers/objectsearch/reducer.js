import { omit } from 'lodash';
import { getPath } from '../helper';
import MusitObject from '../../models/object';
import * as types from './constants';

const initialState = {
  data: {
    totalMatches: 0,
    matches: []
  },
  params: {
    currentPage: 1,
    perPage: 50
  },
  loaded: true
};

export default (state = initialState, action) => {
  switch(action.type) {
  case types.CHANGE_FIELD:
    return {
      ...state,
      params: {
        ...state.params,
        [action.field]: action.value
      }
    };
  case types.SEARCH_OBJECTS:
    return {
      ...omit(state, 'error'),
      loading: true,
      loaded: false,
      data: {
        totalMatches: 0,
        matches: []
      },
      params: {
        ...state.params,
        currentPage: action.page
      }
    };
  case types.SEARCH_OBJECTS_FAIL:
    return {
      ...state,
      loading: false,
      loaded: false,
      data: [],
      error: action.error
    };
  case types.SEARCH_OBJECTS_SUCCESS:
    return {
      ...omit(state, 'error'),
      loading: false,
      loaded: true,
      data: {
        totalMatches: action.result.totalMatches,
        matches: action.result.matches.map(data => {
          return new MusitObject({
            ...data,
            breadcrumb: getPath(data)
          });
        })
      }
    };
  default:
    return state;
  }
};