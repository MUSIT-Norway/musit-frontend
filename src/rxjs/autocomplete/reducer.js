import { Observable } from 'rxjs/Rx';

export default function createReducer(actions) {
  const initialState = [];
  return Observable.of(() => initialState)
    .merge(
      actions.update$.map((data) => () => data),
      actions.clear$.map(() => () => initialState),
    );
}