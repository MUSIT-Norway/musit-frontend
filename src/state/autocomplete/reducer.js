import { Observable } from 'rxjs/Rx';

export default function createReducer(actions, stateKey = "suggest") {
  const initialState = [];
  const reducer$ = Observable.of(() => initialState)
    .merge(
      actions.update$.map((data) => () => data),
      actions.clear$.map(() => () => initialState),
    );
  return Observable.merge(
    reducer$.map(reducer => [stateKey, reducer]),
  );
}