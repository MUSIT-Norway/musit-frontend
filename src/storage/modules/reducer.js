import { toFrontend } from '../../magasin/modules/mapper';
import * as types from './types';

const initialState = {
  state: {
    environmentRequirement: {},
    environmentAssessment: {},
    securityAssessment: {}
  }
};

const storageUnitContainerReducer = (state = initialState, action = {}) => {
  switch (action.type) {
  case types.INSERT:
    return {
      ...state,
      loading: true
    };
  case types.INSERT_SUCCESS:
    return {
      ...state,
      loading: false,
      loaded: true,
      data: toFrontend(action.result)
    };
  case types.INSERT_FAIL:
    return {
      ...state,
      loading: false,
      loaded: true,
      data: action.error
    };
  case types.UPDATE:
    return {
      ...state,
      loading: false,
      loaded: true,
      data: action.data
    };
  case types.LOAD:
    return {
      ...state,
      loading: true
    };
  case types.LOAD_SUCCESS:
    return {
      ...state,
      loading: false,
      loaded: true,
      data: toFrontend(action.result)
    };
  case types.LOAD_FAIL:
    return {
      ...state,
      loading: false,
      loaded: false,
      error: action.error
    };
  case types.UPDATE_STATE:
    return {
      ...state,
      state: action.data
    };
  case types.CLEAR_STATE:
    return {
      ...state,
      state: {
        environmentRequirement: {},
        environmentAssessment: {},
        securityAssessment: {}
      }
    };
  default:
    return state;
  }
};

export default storageUnitContainerReducer;