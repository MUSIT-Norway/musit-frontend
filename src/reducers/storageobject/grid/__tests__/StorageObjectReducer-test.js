import * as actions from '../index';
import reducer from '../index';
import request from 'superagent';
import nocker from 'superagent-nock';
const nock = nocker(request);
import Config from '../../../../config';

const comingFromBackend = [
  {
    id: 1,
    identifier: {
      museumNo: 'C666',
      subNo: '34'
    },
    displayName: 'Ã˜ks'
  },
  {
    id: 2,
    identifier: {
      museumNo: 'C666',
      subNo: '31'
    },
    displayName: 'Sverd'
  },
  {
    id: 3,
    identifier: {
      museumNo: 'C666',
      subNo: '38'
    },
    displayName: 'Sommerfugl'
  }
];

describe('StorageUnitReducer', () => {
  it('creates LOAD_SEVERAL_SUCCESS when fetching data has been done', () => {
    const id = 1;
    const url = `${Config.magasin.urls.thingaggregate.baseUrl(99)}/node/${id}/objects`;
    nock('http://localhost')
        .get(url)
        .reply(200, comingFromBackend);
    const store = mockStore();

    return store.dispatch(actions.loadObjects(1))
        .then(() => {
          expect(store.getActions()).toMatchSnapshot();
        });
  });

  it('no action', () => {
    expect(
        reducer(undefined, undefined)
    ).toMatchSnapshot();
  });

  it('initial action', () => {
    expect(
        reducer(undefined, {
          type: actions.LOAD_SEVERAL
        })
    ).toMatchSnapshot();
  });

  it('success action', () => {
    expect(
        reducer(undefined, {
          type: actions.LOAD_SEVERAL_SUCCESS,
          result: {
            someField: 1
          }
        })
    ).toMatchSnapshot();
  });

  it('fail action', () => {
    expect(
        reducer(undefined, {
          type: actions.LOAD_SEVERAL_FAIL,
          error: Error('Some error here.')
        })
    ).toMatchSnapshot();
  });

});
