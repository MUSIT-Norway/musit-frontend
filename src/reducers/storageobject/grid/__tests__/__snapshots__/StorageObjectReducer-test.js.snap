exports[`StorageUnitReducer creates LOAD_SEVERAL_SUCCESS when fetching data has been done 1`] = `
Array [
  Object {
    "type": "musit/storageobject-grid/LOAD_SEVERAL"
  },
  Object {
    "result": Array [
      Object {
        "displayName": "Ã˜ks",
        "id": 1,
        "identifier": Object {
          "museumNo": "C666",
          "subNo": "34"
        }
      },
      Object {
        "displayName": "Sverd",
        "id": 2,
        "identifier": Object {
          "museumNo": "C666",
          "subNo": "31"
        }
      },
      Object {
        "displayName": "Sommerfugl",
        "id": 3,
        "identifier": Object {
          "museumNo": "C666",
          "subNo": "38"
        }
      }
    ],
    "type": "musit/storageobject-grid/LOAD_SEVERAL_SUCCESS"
  }
]
`;

exports[`StorageUnitReducer fail action 1`] = `
Object {
  "data": Array [],
  "loaded": false,
  "loading": false
}
`;

exports[`StorageUnitReducer initial action 1`] = `
Object {
  "loading": true
}
`;

exports[`StorageUnitReducer no action 1`] = `Object {}`;

exports[`StorageUnitReducer success action 1`] = `
Object {
  "data": Array [
    MusitObject {
      "breadcrumb": undefined,
      "currentLocationId": undefined,
      "id": 1,
      "mainObjectId": undefined,
      "museumNo": "CH500",
      "path": undefined,
      "pathNames": undefined,
      "subNo": "5",
      "term": "Fugl"
    }
  ],
  "loaded": true,
  "loading": false
}
`;
